server/lib/manifest.py

```diff
@@ -0,0 +1,382 @@
+                "Description": "A federated data network allowing access to science data"
+            },
+        "Globus":
+            {
+                "@id": "https://www.materialsdatafacility.org/",
+                "@type": "Organization",
+                "legalName": "Materials Data Facility",
+                "Description": "A simple way to publish, discover, and access materials datasets"
+            }
+    }
+
+    def create_basic_attributes(self):
+        """
+        Returns a portion of basic attributes in the manifest
+        :return: Basic information about the Tale
+        """
+
+        return {
+            "@id": 'https://data.wholetale.org/api/v1/tale/' + str(self.tale['_id']),
+            "createdOn": str(self.tale['created']),
+            "schema:name": self.tale['title'],
+            "schema:description": self.tale.get('description', str()),
+            "schema:category": self.tale['category'],
+            "schema:identifier": str(self.tale['_id']),
+            "schema:version": self.tale['format'],
+            "schema:image": self.tale['illustration'],
+            "aggregates": list(),
+            "Datasets": list()
+        }
+
+    def add_tale_creator(self):
+        """
+        Adds basic information about the Tale author
+        """
+
+        tale_user = self.userModel.load(self.tale['creatorId'], user=self.user)
+        self.manifest['createdBy'] = {
+            "@id": self.tale['authors'],
+            "@type": "schema:Person",
+            "schema:givenName": tale_user.get('firstName', ''),
+            "schema:familyName": tale_user.get('lastName', ''),
+            "schema:email": tale_user.get('email', '')
+        }
+
+    def create_context(self):
+        """
+        Creates the manifest namespace. When a new vocabulary is used, it shoud
+        get added here.
+        :return: A structure defining the used vocabularies
+        """
+        return {
+            "@context": [
+                "https://w3id.org/bundle/context",
+                {"schema": "http://schema.org/"}
+            ]
+        }
+
+    def create_dataset_record(self, folder_id):
+        """
+        Creates a record that describes a Dataset
+        :param folder_id: Folder that represents a dataset
+        :return: Dictionary that describes a dataset
+        """
+        folder = self.folderModel.load(folder_id,
+                                       user=self.user,
+                                       exc=True,
+                                       level=AccessType.READ)
+        try:
+            meta = folder.get('meta')
+            provider = meta.get('provider')
+            return {
+                "@id": meta.get('identifier'),
+                "@type": "Dataset",
+                "name": folder['name'],
+                "identifier": meta.get('identifier'),
+                "publisher": self.publishers[provider]}
+
+        except (KeyError, ValidationException):
+            pass
```

Comment:
This is not exactly what I had in mind. Take look at this:

```
try:
    folder = self.folderModel.load(folder_id, user=self.user, exc=True, level=AccessType.READ)
    provider = folder['meta']['provider']
    identifier = folder['meta']['identifier']
    return {
        "@id": identifier,
        ...
        "publisher": self.publishers[provider]
    }
except (KeyError, TypeError, ValidationException):
     pass
```

1. `Folder().load()` needs to be under `try/except` to catch `ValidationException`
1. Unless you explicitly try to access dictionary via key, `KeyError` will never get raised (can't use `.get()`)

Reply:
Oops that was some overlooking on my part; Fixed in the latest commit.